### PR TITLE
Fixes #173: Allow long keys when looking up points

### LINKED ISSUE #173: tvtk.Points ``__getitem__`` does not accept python long integers
The following example demonstrates the issue

```
In [10] points = tvtk.Points()

In [11]: points.insert_next_point((0,1,2))
Out[11]: 0L

In [12]: points[0]
Out[12]: (0.0, 1.0, 2.0)

In [13]: points[0L]  # raises TypeError
```

### PR DESCRIPTION
**Pull request by dmsurti**
Fixes #173.

The special code generation in wrapping code is modified to allow look up with `long` keys in addition to existing int.

Test case added to test look up with `int`, `long` keys and also `get_point`.

---
Travis may fail for `qt4` with object_cache error, known issue and `wx` for wx src download problems. See PR #221 and #222.

### CODE REVIEW

File: tvtk/special_gen.py [hunk 1]
```diff
@@ -319,8 +319,11 @@ def __iter__(self):
                 yield obj.GetPoint(i)
 
         def _check_key(self, key, n):
-            if type(key) != type(1):
-                raise TypeError, "Only integers are valid keys."
+            ##############################################
+            # Allow int and long keys. Fixes GH Issue 173.
+            ##############################################
+            if type(key) not in [type(1), type(1L)]:
```
Comment (itziakos) [inline tvtk/special_gen.py]: I do not think it is necessary to check for type. We should just propagate the key in this case and let the vtk python to coerce the value and fail (if necessary).

Comment (itziakos) [inline tvtk/special_gen.py]: If key is not a numeric value it will fail anyway in the comparisons that follow.

Comment (dmsurti) [inline tvtk/special_gen.py]: @prabhuramachandran Can you tell why the `_check_key` calls have been added? Was that necessary for VTK < 5.10? Is it safe to drop these calls, as they are also used for IdList and DataArray?

Comment (prabhuramachandran) [inline tvtk/special_gen.py]: IIRC, the main reason for `_check_key` is to support negative indices and raise the correct exceptions where necessary.  VTK will not raise any exceptions for those cases, which could lead to a segfault.  Try this:  
```
import vtk
d = vtk.vtkDoubleArray()
d.SetNumberOfTuples(5)
d.GetTuple(-1)
d.GetTuple(100)
```
You might get a segfault!  The integer/long checks are not technically necessary but should not be harmful.  VTK will raise the exception but the user may be confused when they see the exception so I think I caught it earlier.

Comment (dmsurti) [inline tvtk/special_gen.py]: So is it fine to retain the `_check_key` call?

Comment (itziakos) [inline tvtk/special_gen.py]: keeping the `_check_key` call is ok for know, but we should rework the method to only check the for negative numbers and not the type anymore.

Comment (itziakos) [inline tvtk/special_gen.py]: probably use `isinstance` here

**Comment by itziakos** [comment]
:+1:

**Pull request merged**